### PR TITLE
Removing Awaits

### PR DESCRIPTION
**Pull request by griffhun**
- [x] Removing processing Await from AmqpConsumer
- [x] Remove Await from InMemoryMessageBufferDecorator
- [x] Remove Await from MessageBufferProcessor

Asana Task:
https://app.asana.com/0/0/811418614739943/f

### CODE REVIEW

File: src/main/scala/com/kinja/amqp/AmqpConsumer.scala [hunk 1]
```diff
@@ -87,6 +85,27 @@ class AmqpConsumer(
 	}
 }
 
+object Listener {
+
+	case object WakeUp
+
+	sealed trait ProcessingResult {
+		val originalSender: ActorRef
+		val envelope: Envelope
```
Comment (ghost) [inline src/main/scala/com/kinja/amqp/AmqpConsumer.scala]: Why not use `def` here? You won't need an `override` later.

Comment (ghost) [reply src/main/scala/com/kinja/amqp/AmqpConsumer.scala]: Actually, I don't think you need those at all; see below.

File: src/main/scala/com/kinja/amqp/AmqpConsumer.scala [hunk 2]
```diff
@@ -113,47 +130,82 @@ class Listener[A: Reads](
...
-							context.become(asleep(sender, ack))
-						} else {
-							sender ! ack
-						}
-					} catch {
-						case NonFatal(t) =>
-							logger.warn(s"""[RabbitMQ] Exception while processing message "$s" : $t""")
-							sender ! Reject(envelope.getDeliveryTag, requeue = true)
-					}
+					context.become(processing, true)
```
Comment (ghost) [inline src/main/scala/com/kinja/amqp/AmqpConsumer.scala]: Probably would be better to have `originalSender` and `envelope` as parameters to `processing`. Then you don't need them in the processing result message, and it becomes much easier.

File: src/main/scala/com/kinja/amqp/AmqpConsumer.scala (hunk 1 above)
Comment (ghost) [reply src/main/scala/com/kinja/amqp/AmqpConsumer.scala]: And, thirdly, I don't see why you would need such a trait at all.

File: src/main/scala/com/kinja/amqp/AmqpConsumer.scala [hunk 3]
```diff
@@ -113,47 +130,82 @@ class Listener[A: Reads](
...
+				unstashAll()
+				context.become(sleeping(originalSender, ack))
+			} else {
+				unstashAll()
+				context.become(idle)
+				originalSender ! ack
+			}
+		case Listener.ProcessFailed(originalSender, envelope, messageBody, reason) =>
+			logger.warn(s"""[RabbitMQ] Exception while processing message "$messageBody" : $reason""")
+			originalSender ! Reject(envelope.getDeliveryTag, requeue = true)
```
Comment (ghost) [inline src/main/scala/com/kinja/amqp/AmqpConsumer.scala]: Discussed it offline.

**Review by ghost — APPROVED** [review]
Looks cool.

**Comment by griffhun** [comment]
@migmit-kinja Could you add some thought about last commit?  To be honest I'm not totally satisfied with result because MessageStore API is blocking anyway.

**Comment by ghost** [comment]
TBH, I'm not sure why it has to be.

**Comment by griffhun** [comment]
@migmit-kinja Could you review the last changes I've made?

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**